wherehows-dao/src/main/java/wherehows/dao/table/DatasetComplianceDao.java

```diff
@@ -0,0 +1,154 @@
+   * @param datasetUrn String
+   * @param auditStamp ChangeAuditStamp
+   * @param compliance MCE CompliancePolicy
+   * @throws Exception
+   */
+  public void insertUpdateCompliance(int datasetId, String datasetUrn, ChangeAuditStamp auditStamp,
+      CompliancePolicy compliance) throws Exception {
+
+    // find dataset compliance if exist
+    DsCompliance dsCompliance = null;
+    try {
+      dsCompliance = findBy(DsCompliance.class, "datasetId", datasetId);
+    } catch (Exception e) {
+      log.debug("Can't find dataset compliance for ID: " + datasetId, e.toString());
+    }
+
+    if (dsCompliance == null) {
+      dsCompliance = new DsCompliance();
+      dsCompliance.setDatasetId(datasetId);
+    }
+
+    String actor = getUrnEntity(toStringOrNull(auditStamp.actorUrn));
+    int modifiedTime = (int) (auditStamp.time / 1000);
+
+    fillDsComplianceByCompliancePolicy(dsCompliance, compliance, datasetUrn, actor, modifiedTime);
+
+    update(dsCompliance);
+  }
+
+  /**
+   * Fill in DsCompliance information from MCE CompliancePolicy
+   * @param dsCompliance DsCompliance
+   * @param compliance CompliancePolicy
+   * @param datasetUrn String
+   * @param actor String
+   * @param modifiedTime int
+   */
+  public void fillDsComplianceByCompliancePolicy(DsCompliance dsCompliance, CompliancePolicy compliance,
+      String datasetUrn, String actor, int modifiedTime) {
+
+    dsCompliance.setDatasetUrn(datasetUrn);
+    dsCompliance.setCompliancePurgeType(compliance.compliancePurgeType.name());
+    dsCompliance.setConfidentiality(compliance.datasetConfidentiality.name());
+    dsCompliance.setDatasetClassification(compliance.datasetClassification.toString());
+    if (compliance.complianceEntities != null) {
+      dsCompliance.setComplianceEntities(compliance.complianceEntities.toString());
+    }
+
+    dsCompliance.setModifiedBy(actor);
+    dsCompliance.setModifiedTime(modifiedTime);
+  }
+
+  /**
+   * Insert / update dataset compliance suggestion table given information from MetadataChangeEvent
+   * @param datasetId int
+   * @param datasetUrn String
+   * @param auditStamp ChangeAuditStamp
+   * @param compliance MCE CompliancePolicy
+   * @throws Exception
+   */
+  public void insertUpdateComplianceSuggestion(int datasetId, String datasetUrn, ChangeAuditStamp auditStamp,
+      SuggestedCompliancePolicy compliance) throws Exception {
+
+    // find dataset compliance if exist
+    DsComplianceSuggestion dsCompliance = null;
+    try {
+      dsCompliance = findBy(DsComplianceSuggestion.class, "datasetId", datasetId);
+    } catch (Exception e) {
+      log.debug("Can't find dataset compliance suggestion for ID: " + datasetId, e.toString());
+    }
+
+    if (dsCompliance == null) {
+      dsCompliance = new DsComplianceSuggestion();
+      dsCompliance.setDatasetId(datasetId);
+    }
+
+    String actor = getUrnEntity(toStringOrNull(auditStamp.actorUrn));
+    int modifiedTime = (int) (auditStamp.time / 1000);
```

Comment:
Let's use the row modified time instead of the auditStamp time so it's consistent across all tables

Reply:
Updated to use JPA @preupdate